crypto/pem/pvkfmt.c

```diff
@@ -730,14 +729,15 @@ static EVP_PKEY *do_PVK_body(const unsigned char **in,
                 PEMerr(PEM_F_DO_PVK_BODY, PEM_R_BAD_DECRYPT);
                 goto err;
             }
-        } else
-            OPENSSL_cleanse(keybuf, 20);
+        }
         p = enctmp;
     }
 
     ret = b2i_PrivateKey(&p, keylen);
  err:
     EVP_CIPHER_CTX_free(cctx);
+    if (enctmp != NULL)
+        OPENSSL_cleanse(keybuf, sizeof(keybuf));
```

Comment:
Since you are testing for not NULL, why not push the free into the body of the if?  There isn't any point testing it twice, although it wouldn't hurt if so.

Reply:
Because then all the early exits via goto err cause the same issue (unless you duplicate the cleanse call). It is cleaner to just do the cleanse on exit (the NULL check isnt really needed - it just ensures that it isnt always done)

Reply:
I meant this:

```
if (enctmp != NULL) {
    cleanse
    free
}
```

if enctmp is NULL, the cleanse is skipped and the free does nothing.